src/Symfony/Component/Routing/Tests/Loader/YamlFileLoaderTest.php

```diff
@@ -179,18 +185,88 @@ public function provideFilesImportingRoutesWithControllers()
 
         $this->assertNotNull($routeCollection->get('app_blog'));
         $this->assertEquals('/blog', $routeCollection->get('app_blog')->getPath());
         $this->assertNotNull($routeCollection->get('api_app_blog'));
         $this->assertEquals('/api/blog', $routeCollection->get('api_app_blog')->getPath());
     }
+
+    public function testRemoteSourcesAreNotAccepted()
+    {
+        $this->setupStubbedLoader();
+        $this->expectException(InvalidArgumentException::class);
+        $this->loader->load('http://remote.com/here.yml');
+    }
+
+    public function testLoadingLocalizedRoute()
+    {
+        $routes = $this->loader->load('localized/localized-route.yml');
+
+        $this->assertCount(3, $routes);
+    }
+
+
+    public function testImportingRoutesFromDefinition()
+    {
+        $routes = $this->loader->load('localized/importing-localized-route.yml');
+
+        $this->assertCount(3, $routes);
+        $this->assertEquals('/nl', $routes->get('home.nl')->getPath());
+        $this->assertEquals('/en', $routes->get('home.en')->getPath());
+        $this->assertEquals('/here', $routes->get('not_localized')->getPath());
+    }
+
+    public function testImportingRoutesWithLocales()
+    {
+        $routes = $this->loader->load('localized/importer-with-locale.yml');
+
+        $this->assertCount(2, $routes);
+        $this->assertEquals('/nl/voorbeeld', $routes->get('imported.nl')->getPath());
+        $this->assertEquals('/en/example', $routes->get('imported.en')->getPath());
+    }
+
+    public function testImportingRoutesFromDefinitionMissingLocalePrefix()
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->loader->load('localized/missing-locale-in-importer.yml');
+    }
+
+    public function testImportingNotLocalizedRoutesFromLocalizedImport()
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->loader->load('localized/importing-not-localized-with-localized-prefix.yml');
+    }
+
+    public function testImportingRouteWithoutPathOrLocales()
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->loader->load('localized/route-without-path-or-locales.yml');
+    }
+
+    public function testImportingWithControllerDefault()
+    {
+        $routes = $this->loader->load('localized/importer-with-controller-default.yml');
+        $this->assertCount(3, $routes);
+        $this->assertEquals('DefaultController::defaultAction', $routes->get('home.en')->getDefault('_controller'));
+        $this->assertEquals('DefaultController::defaultAction', $routes->get('home.nl')->getDefault('_controller'));
+
+        /**
+         * Discussion point:
+         * This is current behaviour. When a resource is imported and the defaults are set
+         * the parent defaults have precedence over the imported routes. This seems weird
+         * and probably unwanted to me. This can be fixed by correcting the behaviour in the
```

Comment:
I think this is on purpose: the logic is "who is responsible for their routes?". It must be the consumer - ie the outside of the imported routes. Said differently, imported routes should be overrideable, and this is what the current logic provides.
Yes, applying the same default to all routes might no be best. But then you can still redefine them one by one.
That's my understanding at least.

Reply:
Ah ok, it makes sense when you explain it like that, although I can't imagine anybody really using it like that. Then again, it falls under the BC promise so probably not wise to alter this behaviour. I'll remove the comments and keep the current implementation.